commit 8528e7fcdd11c646ad56ac969168408808ec3746 (refs/changes/80/480/3)
Author: Robert Mustacchi <rm@joyent.com>
Date:   2016-09-22T18:20:15+00:00 (3 years, 1 month ago)
    
    OS-5517 Sun LDAP tools need to support TLS.
    Reviewed by: Jerry Jelinek <jerry.jelinek@joyent.com>
    Reviewed by: Patrick Mooney <patrick.mooney@joyent.com>
    Approved by: Patrick Mooney <patrick.mooney@joyent.com>

diff --git a/nss-nspr/Makefile b/nss-nspr/Makefile
index 92ca373..b006810 100644
--- a/nss-nspr/Makefile
+++ b/nss-nspr/Makefile
@@ -1,31 +1,23 @@
 #
-# CDDL HEADER START
+# This file and its contents are supplied under the terms of the
+# Common Development and Distribution License ("CDDL"), version 1.0.
+# You may only use this file in accordance with the terms of version
+# 1.0 of the CDDL.
 #
-# The contents of this file are subject to the terms of the
-# Common Development and Distribution License, Version 1.0 only
-# (the "License").  You may not use this file except in compliance
-# with the License.
+# A full copy of the text of the CDDL should have accompanied this
+# source.  A copy of the CDDL is also available via the Internet at
+# http://www.illumos.org/license/CDDL.
 #
-# You can obtain a copy of the license at COPYING
-# See the License for the specific language governing permissions
-# and limitations under the License.
-#
-# When distributing Covered Code, include this CDDL HEADER in each
-# file and include the License file at COPYING.
-# If applicable, add the following below this CDDL HEADER, with the
-# fields enclosed by brackets "[]" replaced with your own identifying
-# information: Portions Copyright [yyyy] [name of copyright owner]
-#
-# CDDL HEADER END
+
 #
-# Copyright (c) 2012, Joyent, Inc.  All rights reserved.
+# Copyright 2016 Joyent, Inc.
 #
 
-VER =	nss-3.12.8
+VER =	nss-3.25
 
 include ../Makefile.defs
 
-NSPRVER =	nspr-4.8.6
+NSPRVER =	nspr-4.12
 TARBALL =	$(VER)-with-$(NSPRVER).tar.gz
 
 AUTOCONF_CPPFLAGS =
@@ -40,18 +32,19 @@ include ../Makefile.targ
 all: all_32 all_64
 
 all_32: $(VER.32)/$(UNPACK_SENTINEL)
-	@ (cd $(VER.32)/mozilla/security/nss; \
+	@ (cd $(VER.32)/nss; \
 	    PATH=$(PATH) \
 	    gmake BUILD_OPT=1 BUILD_SUN_PKG=1 NS_USE_GCC=1 NO_MDUPDATE=1 \
 	    CC="$(GCC)" CXX="$(GXX)" CPPFLAGS="$(CPPFLAGS)" \
-	    LDFLAGS="$(LDFLAGS)" nss_build_all )
+	    LDFLAGS="$(LDFLAGS)" NSS_DISABLE_GTESTS=1 nss_build_all )
 
 all_64: $(VER.64)/$(UNPACK_SENTINEL)
-	@ (cd $(VER.64)/mozilla/security/nss; \
+	@ (cd $(VER.64)/nss; \
 	    PATH=$(PATH) \
 	    gmake USE_64=1 BUILD_OPT=1 BUILD_SUN_PKG=1 NS_USE_GCC=1 \
 	    CC="$(GCC.64)" CXX="$(GXX.64)" CPPFLAGS="$(CPPFLAGS)" \
-	    LDFLAGS="$(LDFLAGS.64)" NO_MDUPDATE=1 nss_build_all )
+	    LDFLAGS="$(LDFLAGS.64)" NSS_DISABLE_GTESTS=1 NO_MDUPDATE=1 \
+	    nss_build_all )
 
 install: all
 	DESTDIR=$(DESTDIR) ksh93 ./install-nss $(VER.32)
diff --git a/nss-nspr/Patches/compilers.diff b/nss-nspr/Patches/compilers.diff
index e084b83..2c1eb0b 100644
--- a/nss-nspr/Patches/compilers.diff
+++ b/nss-nspr/Patches/compilers.diff
@@ -1,12 +1,25 @@
-diff -urN nss-3.12.8.orig/mozilla/security/nss/Makefile nss-3.12.8/mozilla/security/nss/Makefile
---- nss-3.12.8.orig/mozilla/security/nss/Makefile	2009-12-08 16:47:03.000000000 +0000
-+++ nss-3.12.8/mozilla/security/nss/Makefile	2014-07-15 16:43:29.856578292 +0000
-@@ -108,7 +108,7 @@
- NSPR_CONFIGURE_OPTS += --enable-debug-rtl
+From e392c6e9f6f656ef4be0c22797d77e4345525202 Mon Sep 17 00:00:00 2001
+From: Robert Mustacchi <rm@joyent.com>
+Date: Tue, 6 Sep 2016 22:56:42 +0000
+Subject: [PATCH] Use correct compiler
+
+---
+ nss/Makefile | 2 +-
+ 1 file changed, 1 insertion(+), 1 deletion(-)
+
+diff --git a/nss/Makefile b/nss/Makefile
+index aa346ae..4800764 100644
+--- a/nss/Makefile
++++ b/nss/Makefile
+@@ -83,7 +83,7 @@ ifdef USE_STATIC_RTL
+ NSPR_CONFIGURE_OPTS += --enable-static-rtl
  endif
  ifdef NS_USE_GCC
--NSPR_COMPILERS = CC=gcc CXX=g++
-+NSPR_COMPILERS = CC="$(CC)" CXX="$(CXX)"
+-NSPR_CONFIGURE_ENV = CC=gcc CXX=g++
++NSPR_CONFIGURE_ENV = CC="$(CC)" CXX="$(CXX)"
  endif
  
- #
+ ifdef SANITIZER_CFLAGS
+-- 
+2.2.1
+
diff --git a/nss-nspr/Patches/sqlite.patch b/nss-nspr/Patches/sqlite.patch
new file mode 100644
index 0000000..58872d4
--- /dev/null
+++ b/nss-nspr/Patches/sqlite.patch
@@ -0,0 +1,10 @@
+--- ignore/nss/lib/sqlite/Makefile.orig	Tue Sep  6 23:47:39 2016
++++ ignore/nss/lib/sqlite/Makefile	Tue Sep  6 23:49:28 2016
+@@ -51,3 +51,7 @@
+ OS_CFLAGS += -w44996
+ endif
+ 
++ifeq (SunOS,$(OS_ARCH))
++# sqlite needs C99 for access to newer SUS standards
++OS_CFLAGS += -std=gnu99
++endif
diff --git a/nss-nspr/install-nss b/nss-nspr/install-nss
index 4ccfba7..ab59637 100644
--- a/nss-nspr/install-nss
+++ b/nss-nspr/install-nss
@@ -1,27 +1,20 @@
 #!/usr/bin/ksh93
 #
-# CDDL HEADER START
+# This file and its contents are supplied under the terms of the
+# Common Development and Distribution License ("CDDL"), version 1.0.
+# You may only use this file in accordance with the terms of version
+# 1.0 of the CDDL.
 #
-# The contents of this file are subject to the terms of the
-# Common Development and Distribution License, Version 1.0 only
-# (the "License").  You may not use this file except in compliance
-# with the License.
+# A full copy of the text of the CDDL should have accompanied this
+# source.  A copy of the CDDL is also available via the Internet at
+# http://www.illumos.org/license/CDDL.
 #
-# You can obtain a copy of the license at COPYING
-# See the License for the specific language governing permissions
-# and limitations under the License.
-#
-# When distributing Covered Code, include this CDDL HEADER in each
-# file and include the License file at COPYING.
-# If applicable, add the following below this CDDL HEADER, with the
-# fields enclosed by brackets "[]" replaced with your own identifying
-# information: Portions Copyright [yyyy] [name of copyright owner]
-#
-# CDDL HEADER END
+
 #
-# Copyright (c) 2010 Joyent Inc., All rights reserved.
+# Copyright 2016 Joyent, Inc.
 #
 
+
 set -o errexit
 
 VERS=${1}
@@ -38,7 +31,7 @@ mkdir -p ${INC_PREFIX}
 
 source ../install.subr
 
-cd ${ROOT}/${VERS}/mozilla/dist/SunOS5.11_i86pc_gcc_OPT.OBJ
+cd ${ROOT}/${VERS}/dist/SunOS5.11_i86pc_gcc_OPT.OBJ
 
 _install N lib/libfreebl3.chk ${PREFIX}/libfreebl3.chk 555
 _install N lib/libfreebl3.so ${PREFIX}/libfreebl3.so 555
@@ -59,6 +52,6 @@ _install N lib/libplds4.so ${PREFIX}/libplds4.so 555
 
 _install N bin/certutil ${BIN_PREFIX}/certutil 555
 
-cp -Lpr ${ROOT}/${VERS}/mozilla/dist/SunOS5.11_i86pc_gcc_OPT.OBJ/include/* \
+cp -Lpr ${ROOT}/${VERS}/dist/SunOS5.11_i86pc_gcc_OPT.OBJ/include/* \
     ${INC_PREFIX}
-cp ${ROOT}/${VERS}/mozilla/dist/public/nss/* ${INC_PREFIX}
+cp ${ROOT}/${VERS}/dist/public/nss/* ${INC_PREFIX}
diff --git a/nss-nspr/install-nss-64 b/nss-nspr/install-nss-64
index 743d66b..761cf9a 100644
--- a/nss-nspr/install-nss-64
+++ b/nss-nspr/install-nss-64
@@ -1,25 +1,17 @@
 #!/usr/bin/ksh93
 #
-# CDDL HEADER START
+# This file and its contents are supplied under the terms of the
+# Common Development and Distribution License ("CDDL"), version 1.0.
+# You may only use this file in accordance with the terms of version
+# 1.0 of the CDDL.
 #
-# The contents of this file are subject to the terms of the
-# Common Development and Distribution License, Version 1.0 only
-# (the "License").  You may not use this file except in compliance
-# with the License.
+# A full copy of the text of the CDDL should have accompanied this
+# source.  A copy of the CDDL is also available via the Internet at
+# http://www.illumos.org/license/CDDL.
 #
-# You can obtain a copy of the license at COPYING
-# See the License for the specific language governing permissions
-# and limitations under the License.
-#
-# When distributing Covered Code, include this CDDL HEADER in each
-# file and include the License file at COPYING.
-# If applicable, add the following below this CDDL HEADER, with the
-# fields enclosed by brackets "[]" replaced with your own identifying
-# information: Portions Copyright [yyyy] [name of copyright owner]
-#
-# CDDL HEADER END
+
 #
-# Copyright (c) 2010 Joyent Inc.
+# Copyright 2016 Joyent, Inc.
 #
 
 set -o errexit
@@ -38,7 +30,7 @@ ln -sf amd64 ${PREFIX}/64
 
 source ../install.subr
 
-cd ${ROOT}/${VERS}/mozilla/dist/SunOS5.11_i86pc_gcc_64_OPT.OBJ
+cd ${ROOT}/${VERS}/dist/SunOS5.11_i86pc_gcc_64_OPT.OBJ
 
 _install N lib/libfreebl3.chk ${PREFIX64}/libfreebl3.chk 555
 _install N lib/libfreebl3.so ${PREFIX64}/libfreebl3.so 555
diff --git a/nss-nspr/nss-3.12.8-with-nspr-4.8.6.tar.gz b/nss-nspr/nss-3.12.8-with-nspr-4.8.6.tar.gz
deleted file mode 100644
index b3c31e7..0000000
Binary files a/nss-nspr/nss-3.12.8-with-nspr-4.8.6.tar.gz and /dev/null differ
diff --git a/nss-nspr/nss-3.25-with-nspr-4.12.tar.gz b/nss-nspr/nss-3.25-with-nspr-4.12.tar.gz
new file mode 100644
index 0000000..3441b30
Binary files /dev/null and b/nss-nspr/nss-3.25-with-nspr-4.12.tar.gz differ
